projects/sbb-esta/angular-business/src/lib/header/navbutton/navbutton.component.ts

```diff
@@ -29,52 +33,67 @@ export class NavbuttonComponent implements AfterViewInit, OnDestroy {
   @ViewChild('button', { static: true }) childButton: ElementRef;
 
   /**
-   * Observer to detect when an eventual dropdown attached to childButton is expanded or collapsed.
-   * Used to propagate change detection.
+   * Dropdown trigger on host, if present, for dropdown interaction and detection.
    */
-  private _attributeObserver: MutationObserver;
+  @ContentChild(DropdownTriggerDirective, { static: true })
+  dropdownTrigger: DropdownTriggerDirective;
 
-  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+  /**
+   * Returns whether childButton's dropdown is expanded.
+   */
+  private _isDropdownExpanded = false;
+  @HostBinding('class.sbb-navbutton-dropdown-expanded')
+  get isDropdownExpanded() {
```

Comment:
If this getter is only necessary for styling, you can use attribute selectors. In this case the current element has the attribute aria-expanded="true" when open.

e.g.
```css
.sbb-navbutton[aria-expanded="true"] { ... }
```